design/ceph/mon-storage-class.md

```diff
@@ -0,0 +1,136 @@
+      # The ceph failure domain will be extracted from the label, which by default is the zone. The nodes running OSDs must have
+      # this label in order for the OSDs to be configured in the correct topology. For topology labels, see
+      # https://rook.io/docs/rook/latest/CRDs/ceph-cluster-crd/#osd-topology.
+      failureDomainLabel: topology.kubernetes.io/zone
+      # The sub failure domain is the secondary level at which the data will be placed to maintain data durability and availability.
+      # The default is "host", which means that each OSD must be on a different node and you would need at least two nodes per zone.
+      # If the subFailureDomain is set to "osd", the OSDs would be allowed anywhere in the same zone including on the same node.
+      # If set to "rack" or some other intermediate failure domain, those labels would also need to be set on the nodes where
+      # the osds are started.
+      subFailureDomain: host
+    zones:
+      - name: us-east-2a
+        arbiter: true
+      - name: us-east-2b
+      - name: us-east-2c
+    volumeClaimTemplate:
+      spec:
+        storageClassName: zone-a-storage
+        resources:
+          requests:
+            storage: 10Gi
+```
+
+The operator will track which mons belong to which zone. The zone assignment for each mon
+will be stored in the `rook-ceph-mon-endpoints` configmap in the same data structure where the host
+assignment is stored for mons with node affinity.
+
+For example, if the zones are called
+"zone-a", "zone-b", and "zone-c", the configmap would contain:
+
+```
+data:
+  data: a=10.99.109.200:6789,b=10.98.18.147:6789,c=10.96.86.248:6789,d=10.96.86.249:6789,e=10.96.86.250:6789
+  mapping: '{"node":{"a":"zone-a","b":"zone-b","c":"zone-c"}}'
+  maxMonId: "3"
+```
+
+If no zones are listed in the `spec.mon.zones`, It will default to arbitrary
+deployment without zonal awareness.
+
+### Mon Failover
+
+Mon failover will replace a mon in the same zone where the previous mon failed.
+This would be accomplished using zone antiaffinity to the mons.
+
+### Mon Storage
+
+The mons can all be backed by a host path or a PVC, similar to any other Rook cluster.
+
+In this example, all the mons will be backed by default volumeClaimTemplate:
+
+```yaml
+ mon:
+    count: 3
+    allowMultiplePerNode: false
+    zones:
+    - name: a
+      volumeClaimTemplate:
+        spec:
+          storageClassName: zone-a-storage
+            resources:
+              requests:
+                storage: 10Gi
+    - name: b
+      volumeClaimTemplate:
+        spec:
+          storageClassName: zone-b-storage
+            resources:
+              requests:
+                storage: 10Gi
+    - name: c
+      volumeClaimTemplate:
+        spec:
+          storageClassName: zone-c-storage
+            resources:
+              requests:
+                storage: 10Gi
+```
+
+Discussion for this design doc can be followed [here](https://github.com/rook/rook/issues/10470).
```

Comment:
I'd suggest the issue can point to this design doc, but no need to reference the other way.
```suggestion
```